Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs

```diff
@@ -72,6 +75,124 @@ public IEnumerator TestPointerDirectionToCameraDirection()
+        /// As GameObjects, pointers can be destroyed at any time. 
+        /// Utilize BaseControllerPointer.DestroyOnSourceLost property to test pointer cache does not break with null references (aka auto-destroyed pointers).
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestDestroyOnSourceLostPointer()
+        {
+            TestButtonUtilities.InstantiateDefaultButton(TestButtonUtilities.DefaultButtonType.DefaultPushButton,
+                out Interactable interactable,
+                out Transform translateTargetObject);
+
+            Vector3 targetStartPosition = translateTargetObject.localPosition;
+
+            // Subscribe to interactable's on click so we know the click went through
+            bool wasClicked = false;
+            interactable.OnClick.AddListener(() => { wasClicked = true; });
+
+            var rightHand = new TestHand(Handedness.Right);
+            yield return rightHand.Show(Vector3.right);
+
+            var rightPokePointer = PlayModeTestUtilities.GetPointer<PokePointer>(Handedness.Right);
+            rightPokePointer.DestroyOnSourceLost = true;
+
+            yield return TestButtonUtilities.TestClickPushButton(interactable.transform, targetStartPosition, translateTargetObject);
+
+            Assert.IsTrue(wasClicked);
+            Assert.IsTrue(UnityObjectExtensions.IsNull(rightPokePointer));
+            Assert.IsNull(PlayModeTestUtilities.GetPointer<PokePointer>(Handedness.Right));
+
+            wasClicked = false;
+            yield return TestButtonUtilities.TestClickPushButton(interactable.transform, targetStartPosition, translateTargetObject);
+            Assert.IsTrue(wasClicked);
+        }
+
+        /// <summary>
+        /// Test that buttons still work when pointer cache is disabled. 
+        /// Pointers that do not auto-destroy themselves on source lost should be destroyed by the input device manager creating the pointers
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestDisabledPointerCache()
+        {
+            TestButtonUtilities.InstantiateDefaultButton(TestButtonUtilities.DefaultButtonType.DefaultPushButton,
+                out Interactable interactable,
+                out Transform translateTargetObject);
+
+            Vector3 targetStartPosition = translateTargetObject.localPosition;
+
+            // Subscribe to interactable's on click so we know the click went through
+            bool wasClicked = false;
+            interactable.OnClick.AddListener(() => { wasClicked = true; });
+
+            PlayModeTestUtilities.GetInputSimulationService().EnablePointerCache = false;
+
+            var rightHand = new TestHand(Handedness.Right);
+            yield return rightHand.Show(Vector3.right);
+
+            var rightPokePointer = PlayModeTestUtilities.GetPointer<PokePointer>(Handedness.Right);
+            Assert.IsNotNull(rightPokePointer);
+            Assert.IsFalse(rightPokePointer.DestroyOnSourceLost);
```

Comment:
What is the behavior if `pointer.DestroyOnSourceLost` is true but `PlayModeTestUtilities.GetInputSimulationService().EnablePointerCache` is false? Would a pointer be disabled, but in a cached object list? I would expect the following

When input source is lost....
EnablePointerCache true, DestroySourceLost true => not sure the behavior
EnablePointerCache true, DestroySourceLost false => Pointer gets disabled but moved into the cache
EnablePointerCache false, DestroySourceLost false => Pointer gets disabled
EnablePointerCache false, DestroySourceLost true=> Pointer gets destroyed

Reply:
> EnablePointerCache true, DestroySourceLost true => not sure the behavior

If DestroyOnSourceLost is true, the pointer will be destroyed. Whether the pointer is "added" to the cache or not, depends on how quickly the pointer destroys itself since it's not immediate by Unity. 

Technically it's possible for a pointer to be added when it's about to be destroyed is added, or a pointer that is inactive but is destroyed at some future point while in the cache. 

In these cases, the cache scans itself and clears null entries with CleanActivePointers() or with the stack popping while loop scan